### DISCUSSION TITLE
Equivalent to SWR's mutate?

### DISCUSSION THREAD

**Discussion by RichHorrocks** [wevm/wagmi#509] [Q&A] [answered]
I'm looking at converting my app to use Wagmi instead of SWR. (I've used both vanilla SWR and the eth-swr package.)

I'm looking for a way to mimic the mutate feature of SWR, such that on receiving an event, the `useContractRead()` hook is called again. This is to ensure my front end remains "live", and based on events, rather than based on a heartbeat.

Is there a clean way of doing this?

**Comment by jxom** [comment, marked as answer]
Can you utilise `refetch` in the return values of `useContractRead`, and place that in your event listener?
https://wagmi.sh/docs/hooks/useContractRead#return-value

**Reply by RichHorrocks** [reply]
Perfect - thanks!
